distributor/src/distributor/messenger.bal

```diff
@@ -43,96 +42,99 @@ function getAwaitResultsMessage(string electionCode, string resultType, string r
-//                                         "\'{\"username\":\"myuser\", \"mobile\":\"0771234567\"}\'");
-// }
-
-// # Register recipient in the mobileSubscribers list and persist in the smsRecipients db table.
-// #
-// # + username - The recipient username
-// # + mobileNo - The recipient number
-// # + return - The status of registration or operation error
-// function registerAsSMSRecipient(string username, string mobileNo) returns string|error {
-
-//     if mobileSubscribers.hasKey(username) {
-//         string errMsg = "Registration failed: username:" + username + " is already registered with mobile:" + mobileNo;
-//         log:printError(errMsg);
-//         return error(ERROR_REASON, message = errMsg);
-//     }
-
-//     // Persist recipient number in database
-//     var status = dbClient->update(INSERT_RECIPIENT, username, mobileNo);
-//     if status is error {
-//         log:printError("Failed to persist recipient no in database", status);
-//         return error(ERROR_REASON, message = "Registration failed: username:" + username + " mobile:" + mobileNo
-//                                             + ": " + <string> status.detail()?.message);
-//     }
-//     mobileSubscribers[username] = mobileNo;
-
-//     return "Successfully registered: username:" + username + " mobile:"  + mobileNo;
-// }
-
-// # Unregister recipient from the mobileSubscribers map and remove from the smsRecipients db table.
-// #
-// # + username - The recipient username
-// # + mobileNo - The recipient number
-// # + return - The status of deregistration or operation error
-// function unregisterAsSMSRecipient(string username, string mobileNo) returns string|error {
-
-//     // Remove persisted recipient number from database
-//     var status = dbClient->update(DELETE_RECIPIENT, username, mobileNo);
-//     if status is error {
-//         log:printError("Failed to remove recipient from the database", status);
-//         return error(ERROR_REASON, message = "Unregistration failed: username:" + username + " mobile:" + mobileNo +
-//                                             ": " + <string> status.detail()?.message);
-//     }
-
-//     if mobileSubscribers.hasKey(username) && mobileSubscribers.get(username) == mobileNo {
-//         return "Successfully unregistered: username:" + username + " mobile:" + mobileSubscribers.remove(username);
-//     }
-
-//     string errMsg = "Unregistration failed: No entry found for username:" + username + " mobile:"  + mobileNo;
-//     log:printError(errMsg);
-//     return error(ERROR_REASON, message = errMsg);
-// }
+# Send SMS notification to all the subscribers.
+#
+# + message - The message to send
+# + resultId - The message identification
+function sendSMS(string message, string resultId) {
+    map<string> currentMobileSubscribers = mobileSubscribers;
+    if (currentMobileSubscribers.length() > 0) {
+        log:printInfo("Sending SMS for " + resultId);
+    }
```

Comment:
Shall we change this to

```suggestion
    if (currentMobileSubscribers.length() == 0) {
        return;
    }
    log:printInfo("Sending SMS for " + resultId);
```

That way, we don't even get to the foreach if the map is empty.